### PR TITLE
Reduce image size 1GB -> 0.3GB

### LINKED ISSUE #84: Reduce image size
Just /usr/lib/x86_64-linux-gnu/icinga2/sbin/icinga2 is 282.2 MiB. (?!)

The whole thing: 1GB.

### PR DESCRIPTION
**Pull request by Al2Klimov**
Commit messages and diff(s) should be self-explanatory.

Ex. for 86814d911d423fc399f9956723dd550b1cd1c345, don't ask me or the internet why that even worked before.

### CODE REVIEW

**Comment by julianbrost** [comment]
> Ex. for [86814d9](https://github.com/Icinga/docker-icinga2/commit/86814d911d423fc399f9956723dd550b1cd1c345), don't ask me or the internet why that even worked before.

```
root@58142e2bc807:/# apt-get install libboost-context1.74
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libboost-context1.74-dev' for regex 'libboost-context1.74'
Note, selecting 'libboost-context1.74.0' for regex 'libboost-context1.74'
```

**Comment by Al2Klimov** [comment]
This answers _how_, but now _why_.

**Comment by julianbrost** [comment]
Isn't "because `libboost-context1.74` is interpreted as a regex and thus installed the actually needed package `libboost-context1.74.0` (among others)" also an answer to why it worked?

**Comment by Al2Klimov** [comment]
Well, it works now #85 as should, that's all I want.

File: compile.bash [hunk 1]
```diff
@@ -10,13 +10,16 @@ mkdir build
...
-	-DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_RUNDIR=/run \
+	-DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_RUNDIR=/run -DICINGA2_LTO_BUILD=ON \
 	-DICINGA2_SYSCONFIGFILE=/etc/sysconfig/icinga2 -DICINGA2_WITH_{COMPAT,LIVESTATUS}=OFF ..
 
 make
 make test
 
 make install "DESTDIR=$(pwd)/../icinga2-bin"
+strip -g ../icinga2-bin/usr/lib/*/icinga2/sbin/icinga2
+strip -g ../icinga2-bin/usr/lib/nagios/plugins/check_nscp_api
```
Comment (julianbrost) [inline compile.bash:21]: I find this change surprising. I'd expect this to happen with `CMAKE_BUILD_TYPE=Release` but apparently that's not the case. I've dug around a bit in the git history and found the following:

- https://github.com/Icinga/icinga2/commit/e5fba1d881c7de0b16ddf8fd662edb9699cc8b44 + https://github.com/Icinga/icinga2/issues/958
- https://github.com/Icinga/icinga2/commit/b6220c6ecfd3a64bf73ae5de8ed62f257867b5c0 + https://github.com/Icinga/icinga2/issues/998

So apparently someone felt the need that release builds need debug symbols, but it remains unclear why.

Comment (Al2Klimov) [reply compile.bash:21]: And... ?

Comment (julianbrost) [reply compile.bash:21]: You want to do a change, so it's up to you to provide the reasoning why this change is a good idea.

Comment (Al2Klimov) [reply compile.bash:21]: Well, because CMAKE_BUILD_TYPE=Release doesn’t do what we want? 🤷‍♂️

Comment (julianbrost) [reply compile.bash:21]: * [icinga2/CMakeLists.txt](https://github.com/Icinga/icinga2/blob/4b754637f7ac06328d101c72fd03e897707fd352/CMakeLists.txt#L251-L265): "All builds shall have debug symbols."
* This PR: "I don't think so!"

So far, you didn't provide reasoning why stripping the debug symbols is a good idea. For example, may this change affect the crash log?

Comment (Al2Klimov) [reply compile.bash:21]: All our _builds_, yes. But not all our _installs_. I mean, our package installs come w/o debug info by default. For this there are extra packages which indeed need it. Do you wanna carry those additional 228MB in every Docker install and so nearly double the image size?

Comment (julianbrost) [reply compile.bash:21]: Good point, the regular packages should basically do the same, so debug symbols can't be very important for normal operation. So the only practical implication might be that you can't easily get debug symbols for the container images. (Is there a "Docker way of doing" for this? Would you typically provide a second image for this?)

But still strange to me why the `CMakeLists.txt` is doing that. I don't get it, isn't that precisely the reason why `Release` and `RelWithDebInfo` exist? Do we also want to do something about that over there?

Comment (Al2Klimov) [reply compile.bash:21]: 1. Let’s wait until (unlikely) someone actually complains.
2. Let’s solve one problem at a time.

Comment (julianbrost) [reply compile.bash:21]: > Let’s solve one problem at a time.

https://github.com/Icinga/icinga2/issues/9623

**Review by julianbrost — APPROVED** [review]

**Comment by Al2Klimov** [comment]
CC @martialblog

**Pull request merged**